hs_core/models.py

```diff
@@ -1381,6 +1381,20 @@ def get_crossref_deposit_xml(self, pretty_print=True):
 
         return '<?xml version="1.0" encoding="UTF-8"?>\n' + etree.tostring(ROOT, pretty_print=pretty_print)
 
+    @property
+    def verbose_name(self):
+        from hs_core.hydroshare.utils import get_resource_types
+        res_cls = None
+        for tp in get_resource_types():
+            if self.resource_type == tp.__name__:
+                res_cls = tp
+                break
+        if res_cls is not None:
+            res_obj = res_cls()
+            return res_obj._meta.verbose_name
+        else:
+            return self._meta.verbose_name
```

Comment:
This is mostly for @jbradberry but is a response to several issues in the code as it is written now for discovery. 

First, All true resources are proxy classes of BaseResource. This is so we can do searches over BaseResource and pick up everything. This greatly simplifies the "my resources" page. 

Second, the actual search engine is configured to harvest BaseResource, and not the individual classes. This is to save a lot of time; we will not have to re-index every time we create a new Resource Type. Since all real resources are proxy subclasses, all the data we need is here. All we have to determine is the true subclass and we have enough information to index. 

This has some important consequences. 

- The actual type of the Django object during indexing is BaseResource. 
- The effective type of the resource is stored in the BaseResource object itself in a field resource_type. 
- To get to the actual proxy object and its methods, one must convert the string resource_type into a real class instance

This complex dance is why get_content_type doesn't work. It returns BaseResource. We need it to return the proxy subclass.